backend/clickhouse/traces.go

```diff
@@ -274,10 +286,128 @@ func (client *Client) ReadTrace(ctx context.Context, projectID int, traceID stri
 	return traces, rows.Err()
 }
 
+func (client *Client) ReadTracesMetrics(ctx context.Context, projectID int, params modelInputs.QueryInput, metricTypes []modelInputs.TracesMetricType, nBuckets int) (*modelInputs.TracesMetrics, error) {
+	startTimestamp := uint64(params.DateRange.StartDate.Unix())
+	endTimestamp := uint64(params.DateRange.EndDate.Unix())
+	useSampling := params.DateRange.EndDate.Sub(params.DateRange.StartDate) >= time.Hour
+
+	fnStr := ""
+	for _, metricType := range metricTypes {
+		switch metricType {
+		case modelInputs.TracesMetricTypeCount:
+			if useSampling {
+				fnStr += ", round(count() * any(_sample_factor))"
+			} else {
+				fnStr += ", toFloat64(count())"
+			}
+		case modelInputs.TracesMetricTypeP50:
+			fnStr += ", quantile(.5)(Duration)"
```

Comment:
good point, I think this could be extended in the future, it would also require storing `TraceAttribute['BatchSize']` as a number so that might warrant a discussion in the future (related to the task for supporting greater than / less than queries)